src/libexpr/eval.hh

```diff
@@ -733,7 +802,18 @@ struct EvalSettings : Config
 
     Setting<Strings> nixPath{
         this, getDefaultNixPath(), "nix-path",
-        "List of directories to be searched for `<...>` file references."};
+        R"(
+          List of directories to be searched for `<...>` file references
+
+          In particular, outside of [pure evaluation mode](#conf-pure-evaluation), this is used to define
+          [`builtins.nixPath`](@docroot@/language/builtin-constants.md#builtin-constants-nixPath).
+
+          See
+          [`builtins.nixPath`](@docroot@/language/builtin-constants.md#builtin-constant    s-nixPath)
+          and
+          [`builtins.findFile`](@docroot@/language/builtins.html#builtins-findFile)
+          for futher details.
```

Comment:
```suggestion
```

Please let's not pile up more things that were not there before. We could have merged this PR a long time ago because all it really does (and IMHO should do) is add mechanism. We can work on additional content item-wise.